From cdfc30e8583960c76ae5abb53ae14625988437fa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Lef=C3=A8vre?= <rlefevre@dmy.fr>
Date: Tue, 10 Dec 2019 14:52:35 +0100
Subject: testsuite: Skip broken tests on Alpine

- encoding004 because Alpine does not have locales
- T2615 (the linker script does not not seem to be correctly interpreted)
- T10458
---
 libraries/base/tests/IO/all.T          | 4 +++-
 testsuite/tests/ghci/linking/dyn/all.T | 1 +
 testsuite/tests/rts/all.T              | 2 ++
 3 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/libraries/base/tests/IO/all.T b/libraries/base/tests/IO/all.T
index dba0e5e3d7..b220fd8b8f 100644
--- a/libraries/base/tests/IO/all.T
+++ b/libraries/base/tests/IO/all.T
@@ -115,7 +115,9 @@ test('encoding001', [], compile_and_run, [''])
 
 test('encoding002', normal, compile_and_run, [''])
 test('encoding003', normal, compile_and_run, [''])
-test('encoding004', extra_files(['encoded-data/']), compile_and_run, [''])
+test('encoding004',
+    [when(platform('x86_64-alpine-linux'), skip),
+     extra_files(['encoded-data/'])], compile_and_run, [''])
 test('encoding005', normal, compile_and_run, [''])
 
 test('environment001', [], run_command,
diff --git a/testsuite/tests/ghci/linking/dyn/all.T b/testsuite/tests/ghci/linking/dyn/all.T
index f8679bcbfe..7b453a7086 100644
--- a/testsuite/tests/ghci/linking/dyn/all.T
+++ b/testsuite/tests/ghci/linking/dyn/all.T
@@ -27,6 +27,7 @@ test('T10955dyn', [extra_files(['A.c', 'B.c'])], run_command,
 test('T10458',
      [extra_files(['A.c']),
       unless(doing_ghci, skip),
+      when(platform('x86_64-alpine-linux'), skip),
       pre_cmd('$MAKE -s --no-print-directory compile_libT10458'),
       extra_hc_opts('-L"$PWD/T10458dir" -lAS')],
      ghci_script, ['T10458.script'])
diff --git a/testsuite/tests/rts/all.T b/testsuite/tests/rts/all.T
index a180491f77..8b623e592b 100644
--- a/testsuite/tests/rts/all.T
+++ b/testsuite/tests/rts/all.T
@@ -127,6 +127,8 @@ test('T2615',
        # Solaris' linker does not support GNUish linker scripts
        when(opsys('solaris2'), skip),
        pre_cmd('$MAKE -s --no-print-directory T2615-prep'),
+       # Alpine linker does not recognize exec format
+       when(platform('x86_64-alpine-linux'), skip),
        # Add current directory to dlopen search path
        cmd_prefix('LD_LIBRARY_PATH=$LD_LIBRARY_PATH:. '),
        extra_clean(['libfoo_T2615.so', 'libfoo_T2615.o'])],
-- 
2.17.1

